src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstExistsExpression.cs

```diff
@@ -0,0 +1,61 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Linq.Linq3Implementation.Ast.Visitors;
+
+namespace MongoDB.Driver.Linq.Linq3Implementation.Ast.Expressions
+{
+    internal sealed class AstExistsExpression : AstExpression
```

Comment:
Anyone remember "snigglets"? (Words that should be in the dictionary but aren't)

This is an operator that should exist in the server but doesn't.

But semantically it makes sense for us to pretend it exists, and then we can later simplify/render it into something the server understands.